frontend/components/SettingsPage/SettingsDrawer.tsx

```diff
@@ -0,0 +1,214 @@
+
+const { Text } = Typography;
+
+interface SettingsDrawerProps {
+  isDrawerOpen: boolean;
+  onClose: () => void;
+}
+
+interface TableData extends Record<string, unknown> {
+  key: string;
+  chain: string;
+  threshold: string;
+  topUpAmount: string;
+  tokenSymbol: string;
+}
+
+const SettingsDrawerTitle = (
+  <Flex vertical gap={8}>
+    <Text strong className="text-lg">
+      Default Settings
+    </Text>
+    <Text type="secondary" className="text-sm">
+      Predefined system values, for reference only.
+    </Text>
+  </Flex>
+);
+
+export const SettingsDrawer = ({
+  isDrawerOpen,
+  onClose,
+}: SettingsDrawerProps) => {
+  const { masterEoa, getMasterSafeOf } = useMasterWalletContext();
+  const { selectedAgentConfig, services } = useServices();
+  const { data: settings, isLoading } = useSettingsDrawer();
+  const masterSafe = getMasterSafeOf
+    ? getMasterSafeOf(selectedAgentConfig.evmHomeChainId)
+    : undefined;
+
+  const tableData = useMemo<TableData[]>(() => {
+    if (!settings?.eoa_topups || !services) return [];
+
+    const activeChains = new Set(
+      services.map((service) => String(service.home_chain)),
+    );
+
+    return Object.values(CHAIN_CONFIG)
+      .filter((chainConfig) => {
+        // Only show chains that have active services
+        return activeChains.has(String(chainConfig.middlewareChain));
+      })
+      .map((chainConfig) => {
+        const chainName = chainConfig.name.toLowerCase();
+        const eoaTopups = settings.eoa_topups[chainName];
+        const fundingRequirement = eoaTopups
+          ? Object.values(eoaTopups)[0] || 0
+          : 0;
+
+        // TODO: Replace calculation with threshold value from BE once endpoint is ready
+        const refundingThreshold = fundingRequirement * 0.5;
+
+        return {
+          key: String(chainConfig.evmChainId),
+          chain: chainConfig.name,
+          threshold: `${formatUnits(String(refundingThreshold), 18)} ${chainConfig.nativeToken.symbol}`,
+          topUpAmount: `${formatUnits(String(fundingRequirement), 18)} ${chainConfig.nativeToken.symbol}`,
+          tokenSymbol: chainConfig.nativeToken.symbol,
+        };
+      });
+  }, [settings, services]);
+
+  const columns = [
+    {
+      title: 'Chain',
+      key: 'chain',
+      render: (_: unknown, record: TableData) => (
+        <Flex align="center" gap={8}>
+          <Image
+            src={`/chains/${kebabCase(record.chain)}-chain.png`}
+            alt={record.chain}
+            width={20}
+            height={20}
+            className="flex"
+          />
+          <Text>{record.chain}</Text>
+        </Flex>
+      ),
+      width: '33%',
+    },
+    {
+      title: 'Threshold',
+      key: 'threshold',
+      render: (_: unknown, record: TableData) => (
+        <Flex align="center" gap={8}>
+          <Image
+            src={
+              TokenSymbolConfigMap[
+                record.tokenSymbol as keyof typeof TokenSymbolConfigMap
```

Comment:
why we need `as keyof typeof TokenSymbolConfigMap`? I think it should be just.

```suggestion
                record.tokenSymbol
```

Reply:
Doing this creates an error:
```Element implicitly has an 'any' type because expression of type 'any' can't be used to index type 'Record<"ETH" | "OLAS" | "USDC" | "XDAI" | "WXDAI", { image: string; }>'```